src/main/java/io/cryostat/agent/remote/InvokeContext.java

```diff
@@ -0,0 +1,110 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.cryostat.agent.remote;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.management.ManagementFactory;
+import java.lang.management.MemoryMXBean;
+
+import javax.inject.Inject;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.sun.net.httpserver.HttpExchange;
+import org.apache.http.HttpStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class InvokeContext implements RemoteContext {
+
+    private final Logger log = LoggerFactory.getLogger(getClass());
+    private final ObjectMapper mapper;
+
+    @Inject
+    InvokeContext(ObjectMapper mapper) {
+        this.mapper = mapper;
+    }
+
+    @Override
+    public String path() {
+        return "/mbean-invoke/";
+    }
+
+    /*
+    * public <T> T invokeMBeanOperation(
+           String beanName,
+           String operation,
+           Object[] parameters,
+           String[] signature,
+           Class<T> returnType)
+    */
+
+    @Override
+    public void handle(HttpExchange exchange) throws IOException {
+        try {
+            String mtd = exchange.getRequestMethod();
+            switch (mtd) {
+                case "POST":
+                    try (InputStream body = exchange.getRequestBody()) {
+                        MBeanInvocationRequest req =
+                                mapper.readValue(body, MBeanInvocationRequest.class);
+                        if (req.beanName.equals(ManagementFactory.MEMORY_MXBEAN_NAME)) {
+                            MemoryMXBean bean = ManagementFactory.getMemoryMXBean();
+                            bean.gc();
+                            exchange.sendResponseHeaders(HttpStatus.SC_OK, BODY_LENGTH_NONE);
+                        }
+                    } catch (Exception e) {
+                        log.error("mbean serialization failure", e);
+                        exchange.sendResponseHeaders(HttpStatus.SC_BAD_GATEWAY, BODY_LENGTH_NONE);
+                    }
+                    break;
+                default:
+                    log.warn("Unknown request method {}", mtd);
+                    exchange.sendResponseHeaders(
+                            HttpStatus.SC_METHOD_NOT_ALLOWED, BODY_LENGTH_NONE);
+                    break;
+            }
+        } finally {
+            exchange.close();
+        }
+    }
+
+    static class MBeanInvocationRequest<T> {
+
+        public String beanName;
+        public String operation;
+        public Object[] parameters;
+        public String[] signature;
+        public Class<T> returnType;
+
+        // Support GC and Thread operations for now
+        public boolean isValid() {
+            if (this.beanName.equals(ManagementFactory.MEMORY_MXBEAN_NAME)
+                    || this.beanName.equals(ManagementFactory.THREAD_MXBEAN_NAME)) {
```

Comment:
Let's remove the Thread bean for now, since I wasn't able to get anything useful working with that yet.